apps/total-typescript/src/lib/bookmarks.ts

```diff
@@ -0,0 +1,167 @@
+import {z} from 'zod'
+import {prisma} from '@skillrecordings/database'
+
+export const BookmarkInputSchema = z.object({
+  userId: z.string(),
+  resourceId: z.string(),
+  type: z.string(),
+  fields: z.record(z.string().nullable()).optional(),
+})
+
+export const ReturnedBookmarkSchema = z.object({
+  id: z.string(),
+  resourceId: z.string(),
+  type: z.string(),
+  fields: z.record(z.string().nullable()).optional(),
+})
+
+export const ReturnedBookmarkForBookSchema = z.object({
+  id: z.string(),
+  resourceId: z.string(),
+  type: z.string(),
+  fields: z.object({
+    chapterSlug: z.string(),
+    chapterTitle: z.string(),
+    resourceTitle: z.string(),
+    resourceSlug: z.string(),
+  }),
+})
+
+export type Bookmark = z.infer<typeof ReturnedBookmarkSchema>
+export type BookmarkForBook = z.infer<typeof ReturnedBookmarkForBookSchema>
+export type BookmarkInput = z.infer<typeof BookmarkInputSchema>
+
+export async function getBookmarksForUser(
+  userId: string,
+  resourceIds?: string[],
+) {
+  try {
+    const bookmarks = await prisma.bookmark.findMany({
+      select: {
+        id: true,
+        resourceId: true,
+        type: true,
+        fields: true,
+      },
+      where: {
+        userId,
+        ...(resourceIds
+          ? {
+              resourceId: {
+                in: resourceIds,
+              },
+            }
+          : {}),
+      },
+    })
```

Comment:
when getting user bookmarks we pass `resourceIds` as an array and filter for only resources that are in that array. For prisma we have to do this spread syntax because we don't want to set a value if no `resourceIds` are set